launcher/src/main/java/org/wso2/carbon/launcher/extensions/DropinsBundleDeployerUtils.java

```diff
@@ -0,0 +1,273 @@
+    private static final String dropinsDirectory = "dropins";
+
+    private static List<BundleInfo> newBundlesInfo;
+
+    /**
+     * Updates the specified Carbon profile's bundles.info file based on the OSGi bundles deployed in the dropins
+     * directory. The OSGi bundle information in the bundles.info file in a Carbon profile is used to install and
+     * start the bundles at the server startup for the particular profile.
+     * <p>
+     * The mechanism used in updating the bundles.info file is as follows:
+     * 1. The new OSGi bundle information from the bundles currently existing within the dropins folder are obtained.
+     * The new OSGi bundle information are read only once for updating one or more Carbon profiles.
+     * 2. The existing OSGi dropins bundle information are compared with the newly retrieved bundle information and
+     * the bundles.info file is updated only if the new bundle information are different from the existing.
+     * 3. The new OSGi bundle information replace the existing dropins bundle information from the bundles.info file.
+     * The OSGi bundle information of the non-dropins bundles, retrieved from the bundles.info file and the new dropins
+     * OSGi bundle information are merged together.
+     * 4. Updates the bundles.info file with the OSGi bundle information retrieved in step 3.
+     *
+     * @param carbonHome    the {@link String} representation of carbon.home
+     * @param carbonProfile the bundles.info file to be updated
+     * @throws IOException if an I/O error occurs
+     */
+    public static synchronized void executeDropinsCapability(String carbonHome, String carbonProfile)
+            throws IOException {
+        Path dropinsDirectoryPath = Paths.get(carbonHome, Constants.OSGI_REPOSITORY, dropinsDirectory);
+        Path bundlesInfoFile = Paths.
+                get(carbonHome, Constants.OSGI_REPOSITORY, Constants.PROFILE_PATH, carbonProfile, "configuration",
+                        "org.eclipse.equinox.simpleconfigurator", "bundles.info");
+
+        if (newBundlesInfo == null) {
+            newBundlesInfo = getNewBundlesInfo(dropinsDirectoryPath);
+        }
+
+        if (hasToUpdateBundlesInfoFile(newBundlesInfo, bundlesInfoFile)) {
+            logger.log(Level.INFO, "New file changes detected in " + dropinsDirectory);
+
+            List<BundleInfo> effectiveNewBundleInfo = mergeDropinsBundleInfo(newBundlesInfo, bundlesInfoFile);
+
+            logger.log(Level.INFO, "Updating the OSGi bundle information of Carbon Profile: " + carbonProfile + "...");
+            updateBundlesInfo(effectiveNewBundleInfo, bundlesInfoFile);
+            logger.log(Level.INFO,
+                    "Successfully updated the OSGi bundle information of Carbon Profile: " + carbonProfile);
+        } else {
+            logger.log(Level.INFO, "No changes detected in the dropins directory, skipped the OSGi bundle information "
```

Comment:
As it was discussed few days back, it was agreed to have these printed on the console notifying the updated profile and etc.